New Resource - `azurerm_role_assignment_marketplace` (#22398)
hashicorp/terraform-provider-azurerm: website/docs/r/role_assignment_marketplace.html.markdown

---
subcategory: "Authorization"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_role_assignment_marketplace"
description: |-

>   Assigns a given Principal (User or Group) to a given Role in Private Azure Marketplace.
> ---
> 
> # azurerm_role_assignment_marketplace
> 
> Assigns a given Principal (User or Group) to a given Role in Private Azure Marketplace.

Reviewer: ```suggestion
  Assigns a given Principal (User or Group) to a given Role in a Private Azure Marketplace.
---

# azurerm_role_assignment_marketplace

Assigns a given Principal (User or Group) to a given Role in a Private Azure Marketplace.
```

Author: doc updated

* `role_definition_id` - (Optional) The Scoped-ID of the Role Definition. Changing this forces a new resource to be created. Conflicts with `role_definition_name`.

* `role_definition_name` - (Optional) The name of a built-in Role. Changing this forces a new resource to be created. Conflicts with `role_definition_id`.

~> **NOTE:** To assign `Marketplace Admin` role, the calling Principal must first be assigned Privileged Role Administrator (like `Owner` role) or Global Administrator. See [documentation](https://learn.microsoft.com/en-us/marketplace/create-manage-private-azure-marketplace-new#prerequisites) for more information.

* `condition` - (Optional) The condition that limits the resources that the role can be assigned to. Changing this forces a new resource to be created.

* `condition_version` - (Optional) The version of the condition. Possible values are `1.0` or `2.0`. Changing this forces a new resource to be created.

* `delegated_managed_identity_resource_id` - (Optional) The delegated Azure Resource ID which contains a Managed Identity. Changing this forces a new resource to be created.

> ~> **NOTE:** this field is only used in cross tenant scenario.

Reviewer: ```suggestion
~> **NOTE:** This field is only used in cross tenant scenarios.
```

Author: doc updated

* `description` - (Optional) The description for this Role Assignment. Changing this forces a new resource to be created.

* `skip_service_principal_aad_check` - (Optional) If the `principal_id` is a newly provisioned `Service Principal` set this value to `true` to skip the `Azure Active Directory` check which may fail due to replication lag. This argument is only valid if the `principal_id` is a `Service Principal` identity. Defaults to `false`. Changing this forces a new resource to be created.

> ~> **NOTE:** If it is not a `Service Principal` identity it will cause the role assignment to fail.

Reviewer: This note isn't clear on what we're expecting from the user - is this note saying that the `principal_id` cannot be set to any other type other than a Service Principal? Or should `skip_service_principal_aad_check` always be set to `false` if `principal_id` is something other than a Service Principal?
```suggestion
~> **NOTE:** If it is not a `Service Principal` identity it will cause the role assignment to fail.
```

Author: doc updated: This field takes effect only when `principal_id` is a `Service Principal` identity.

## Attributes Reference

In addition to the Arguments listed above - the following Attributes are exported:

* `id` - The Role Assignment ID.

* `principal_type` - The type of the `principal_id`, e.g. User, Group, Service Principal, Application, etc.

## Timeouts

The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/language/resources/syntax#operation-timeouts) for certain actions:

* `create` - (Defaults to 30 minutes) Used when creating the Role Assignment.
* `update` - (Defaults to 30 minutes) Used when updating the Role Assignment.
* `read` - (Defaults to 5 minutes) Used when retrieving the Role Assignment.
* `delete` - (Defaults to 30 minutes) Used when deleting the Role Assignment.

## Import

Role Assignments can be imported using the `resource id`, e.g.

```shell
terraform import azurerm_role_assignment_marketplace.example /providers/Microsoft.Marketplace/providers/Microsoft.Authorization/roleAssignments/00000000-0000-0000-0000-000000000000
```

> ~> **NOTE:** for cross tenant scenario, the format of `resource id` is composed of Azure resource ID and tenantId. for example:

Reviewer: ```suggestion
~> **NOTE:** For cross tenant scenarios, the format of the `resource id` consists of the Azure resource ID and and the tenant ID, for example:
```

Author: doc updated
